app/frontend/src/features/connections/friends/useFriendList.test.tsx

```diff
@@ -86,39 +84,31 @@ describe("when the listFriends query succeeds", () => {
     getUserMock.mockImplementation((userId: string) => {
       return userId === "2"
         ? getUser(userId)
-        : new Promise((r) => setTimeout(() => r(getUser(userId)), 1));
+        : new Promise((r) => setTimeout(() => r(getUser(userId)), 50));
     });
 
     const { result, waitForNextUpdate } = renderHook(() => useFriendList(), {
       wrapper,
     });
-    await waitForNextUpdate();
+    await waitForNextUpdate({ timeout: 10 });
```

Comment:
Was this necessary to not fail? I'd rather avoid waiting in tests. Not sure if that's possible though.

Reply:
Yeah that wait can't be avoided, since any promise based ops (even ones that have already resolved) won't get its result until the next tick of the event loop, so the wait lets this run for one tick. It's not really a problem if the promise will resolve straight away - most of these tests take like ms to run.